### ISSUE TITLE
is the Ferret workaround in MOM_grid_initialize still necessary?

### ISSUE THREAD

**Issue by wrongkindofdoctor** [NOAA-GFDL/MOM6#992]
@marshallward 
I am replacing the `MOM_read_data` calls with the new fms-io `read_data` calls, but am running into issues with the reconciling the size of the temporary variable allocated to hold the  'x' and 'y' data that are then used to derive the geoLat and geoLon coordinates in [set_grid_metrics_from_mosaic](https://github.com/NOAA-GFDL/MOM6/blob/1f790eb874e5ae03fdaad8b66a51d73faaa9e1c2/src/initialization/MOM_grid_initialize.F90#L369).

```
 ! Construct axes for diagnostic output (only necessary because "ferret" uses
  ! broken convention for interpretting netCDF files).
  start(:) = 1 ; nread(:) = 1
  start(2) = 2 ; nread(1) = ni+1 ; nread(2) = 2
  allocate( tmpGlbl(ni+1,2) )
  if (is_root_PE()) &
    call read_data(filename, "x", tmpGlbl, start, nread, no_domain=.TRUE.)
  call broadcast(tmpGlbl, 2*(ni+1), root_PE())
2*(j-G%jsg)+3
  ! I don't know why the second axis is 1 or 2 here. -RWH
  do i=G%isg,G%ieg
    G%gridLonT(i) = tmpGlbl(2*(i-G%isg)+2,2)
  enddo
  ! Note that the dynamic grid always uses symmetric memory for the global
  ! arrays G%gridLatB and G%gridLonB.
  do I=G%isg-1,G%ieg
    G%gridLonB(I) = tmpGlbl(2*(I-G%isg)+3,1)
  enddo
  deallocate( tmpGlbl )

  allocate( tmpGlbl(1, nj+1) )
  start(:) = 1 ; nread(:) = 1
  start(1) = int(ni/4)+1 ; nread(2) = nj+1
  if (is_root_PE()) &
    call read_data(filename, "y", tmpGlbl, start, nread, no_domain=.TRUE.)
  call broadcast(tmpGlbl, nj+1, root_PE())

  do j=G%jsg,G%jeg
    G%gridLatT(j) = tmpGlbl(1,2*(j-G%jsg)+2)
  enddo
  do J=G%jsg-1,G%jeg
    G%gridLatB(J) = tmpGlbl(1,2*(j-G%jsg)+3)
  enddo
  deallocate( tmpGlbl )
```
Here, the `nread` values are set to _ni+1_ and _nj+1_--one greater than the global domain sizes because of the symmetric memory. 

The new IO fails because it only reads up to the maximum value of the global domain.

_git blame_ shows that this code was inserted 7 years ago as a workaround for ferret. Do any of the MOM developers know if the problem has  been fixed in ferret, or if this section of the code is still necessary either way?

Otherwise, would it be sufficient to read in the global data as to the tmp array sized (niglobal, njglobal), and set last B grid Lon and Lat values equal to the first values? For example, something like:

```
start(:) = 1 ; nread(:) = 1
  start(2) = 2 ; nread(1) = ni ; nread(2) = 2
  allocate( tmpGlbl(ni+1,2) )
! new IO read call  
call read_data(fileObjRead, 'x', tmpGlbl,corner=start(1:2), edge_lengths=nread(1:2))
 ...
  ! Note that the dynamic grid always uses symmetric memory for the global
  ! arrays G%gridLatB and G%gridLonB.

  ! The global arrays returned by new io do not have the extra indices that match ni+1 and nj+1
  ! assume first and last indices have same data to define end points on the B grid
  do I=G%isg-1,G%ieg-1
    G%gridLonB(I) = tmpGlbl(2*(I-G%isg)+3,1)
  enddo
  G%gridLonB(G%ieg) =  tmpGlbl(1,1)

```

**Comment by marshallward** [comment]
We discussed this together with @MJHarrison-GFDL and this seems to not necessarily be a Ferret issue, but about supporting axes for symmetric grids in `ocean_hgrid.nc`.

We still want to read in the full axis, since we don't want to assume a periodic grid or otherwise (rigid boundary, OBC, etc), so the simplest path forward is to stick with the current implementation and read it as non-gridded data, and keep the current implementation.

We can think about improving symmetric grid support or streamlining the MPI operations in the future, and focus for now on porting the existing code to the new FMS IO library.

**Comment by marshallward** [comment]
Is it OK to close this issue?  Most of the specific issues would be addressed in #1033.
